### ISSUE TITLE
ambiguous memoization example in README

### ISSUE THREAD

**Issue by Dowwie** [rust-lang/regex#556]
Would you consider a less ambiguous example literal to use than "..."?

https://play.integer32.com/?version=stable&mode=debug&edition=2018&gist=c6630a07c7a64a81b671deab667d684e

```rust
#[macro_use] extern crate lazy_static;
extern crate regex;

use regex::Regex;

fn some_helper_function(text: &str) -> bool {
    lazy_static! {
        static ref RE: Regex = Regex::new("...").unwrap();
    }
    RE.is_match(text)
}


fn main() {
    println!("bla: {}", some_helper_function("bla"));
    println!("bla...: {}", some_helper_function("bla..."));   
}
```

Result:
```bash
bla: true
bla...: true
```

**Comment by BurntSushi** [comment]
I don't understand what you're requesting. I don't understand why you think `...` is ambiguous and why it's a bad thing. The purpose of that example has nothing to do with the specific regex pattern.

**Comment by Dowwie** [comment]
Yes, it's showing a memoization technique (and one I'm now using!).

Understand that not everyone looks at regex patterns as in depth as you do.  I experiment with patterns until I find one that works and move on, not touching regex for months at a time.  I didn't know until I ran this README example that a regex compiled for "..." would return true on every example I passed to it.

I guess that was what you intended..

**Comment by BurntSushi** [comment]
> Understand that not everyone looks at regex patterns as in depth as you do.

I understand that perfectly well, thank you. That's one of the reasons why I've spent so much time writing documentation.

> I didn't know until I ran this README example that a regex compiled for "..." would return true on every example I passed to it.

The regex `...` means, "match any 3 characters, except for `\n`." So if you give it a string that contains fewer than 3 characters (or no sequence of 3 consecutive non-`\n` characters), then you won't find a match. [Example.](https://play.integer32.com/?version=stable&mode=debug&edition=2018&gist=da0af87ad49a27dd9e329c5a178b1480)

Given that the specific regex pattern in the example isn't really the point, I'm inclined to leave it as is. When I originally wrote the example, I intended `...` to mean "insert whatever here," e.g., an ellipsis.
